datafusion/physical-plan/src/topk/mod.rs

```diff
@@ -282,6 +327,54 @@ impl TopKHeap {
         }
     }
 
+    /// Get threshold values for all columns in the given sort expressions.
+    /// If the heap does not yet have k items, returns None.
+    /// Otherwise, returns the threshold values from the max row in the heap.
+    pub fn get_threshold_values(
+        &self,
+        sort_exprs: &[PhysicalSortExpr],
+    ) -> Result<Option<Vec<ColumnThreshold>>> {
+        // If the heap doesn't have k elements yet, we can't create thresholds
+        let max_row = match self.max() {
+            Some(row) => row,
+            None => return Ok(None),
+        };
+
+        // Get the batch that contains the max row
+        let batch_entry = match self.store.get(max_row.batch_id) {
+            Some(entry) => entry,
+            None => return internal_err!("Invalid batch ID in TopKRow"),
+        };
+
+        // Extract threshold values for each sort expression
+        let mut thresholds = Vec::with_capacity(sort_exprs.len());
+        for sort_expr in sort_exprs {
+            // Extract the value for this column from the max row
+            let expr = Arc::clone(&sort_expr.expr);
+            let value = expr.evaluate(&batch_entry.batch.slice(max_row.index, 1))?;
+
+            // Convert to scalar value - should be a single value since we're evaluating on a single row batch
+            let scalar = match value {
+                ColumnarValue::Scalar(scalar) => scalar,
+                ColumnarValue::Array(array) if array.len() == 1 => {
+                    // Extract the first (and only) value from the array
+                    datafusion_common::ScalarValue::try_from_array(&array, 0)?
+                }
+                array => {
+                    return internal_err!("Expected a scalar value, got {:?}", array)
+                }
+            };
+
+            thresholds.push(ColumnThreshold {
+                expr,
+                value: scalar,
+                sort_options: sort_expr.options,
+            });
+        }
+
+        Ok(Some(thresholds))
+    }
```

Comment:
This deserves some scrutiny. I'm not sure if there's a better way to do this, or if it's wrong for some cases, etc.